prov/gni/src/gnix_util.h

```diff
@@ -38,16 +38,23 @@
 #define _GNIX_UTIL_H_
 
 #include <stdio.h>
-#include "fi_log.h"
-
-#define GNIX_DEBUG (2)
+#include "rdma/fi_log.h"
 
 extern const char gnix_fab_name[];
 extern const char gnix_dom_name[];
 
-#define GNIX_LOG_INFO(...) FI_LOG(GNIX_DEBUG, gnix_fab_name, __VA_ARGS__)
-#define GNIX_LOG_WARN(...) FI_WARN(gnix_fab_name, __VA_ARGS__)
-#define GNIX_LOG_ERROR(...) FI_WARN(gnix_fab_name, __VA_ARGS__)
+extern struct fi_provider gnix_prov;
+
+#define GNIX_WARN(subsystem, ...)                                              \
+	FI_WARN(&gnix_prov, subsystem, __VA_ARGS__)
+#define GNIX_TRACE(subsystem, ...)                                             \
+	FI_TRACE(&gnix_prov, subsystem, __VA_ARGS__)
+#define GNIX_INFO(subsystem, ...)                                              \
+	FI_INFO(&gnix_prov, subsystem, __VA_ARGS__)
+#define GNIX_ERR(subsystem, ...)                                               \
+	FI_WARN(&gnix_prov, subsystem, __VA_ARGS__)
+#define GNIX_DEBUG(subsystem, ...)                                             \
+	FI_DBG(&gnix_prov, subsystem, __VA_ARGS__)
```

Comment:
I defined one for each log level including an extra one (ERR) that is the same as GNIX_WARN but will be used to delineate in code where something is an error and not just a warning. Both will get logged regardless. Should we try to wrap the subsystems, or is using FI_LOG_* pre-defined subsystems acceptable?
@sungeunchoi @hppritcha